aten/src/ATen/native/quantized/cpu/qpool.cpp

```diff
@@ -401,6 +401,29 @@ Tensor quantized_max_pool2d(
   return qy;
 }
 
+std::tuple<at::Tensor &,at::Tensor &> max_pool2d_with_indices_out_quantized_cpu(const at::Tensor & self, at::IntArrayRef kernel_size, at::IntArrayRef stride, at::IntArrayRef padding, at::IntArrayRef dilation, bool ceil_mode, at::Tensor & out, at::Tensor & indices) {
+  out = at::native::quantized_max_pool2d(self, kernel_size, stride, padding, dilation, ceil_mode);
+  indices = at::empty(out.sizes(), indices.options());
+  const int kH = safe_downcast<int, int64_t>(kernel_size[0]);
+  const int kW = kernel_size.size() == 1 ? kH : safe_downcast<int, int64_t>(kernel_size[1]);
+
+  const int dH = stride.empty() ? kH : safe_downcast<int, int64_t>(stride[0]);
+  const int dW = stride.empty() ? kW :
+                 stride.size() == 1 ? dH : safe_downcast<int, int64_t>(stride[1]);
+
+  const int padH = safe_downcast<int, int64_t>(padding[0]);
+  const int padW = padding.size() == 1 ? padH : safe_downcast<int, int64_t>(padding[1]);
+
+  const int dilationH = safe_downcast<int, int64_t>(dilation[0]);
+  const int dilationW = dilation.size() == 1 ? dilationH : safe_downcast<int, int64_t>(dilation[1]);
+  // the output tensor is already computed from quantized_max_pool2d. junk_out is used as a dummy
+  // argument because it is required by max_pool2d_kernel
+  auto junk_out = at::empty(out.sizes(), self.int_repr().options());
+  max_pool2d_kernel(kCPU, junk_out, indices, self.int_repr(), kW, kH, dW, dH, padW, padH, dilationW, dilationH);
```

Comment:
we should be able to get the quantized output (quint8) based on the output of max_pool2d_kernel (uint8) I think, the output of max_pool2d_kernel will be the int_repr of the actual quantized output, we can make a quantized tensor with `_make_per_tensor_quantized_tensor `

Reply:
Yes, the code as written looks awful, you're running max_pool2d twice to get out the quantized output and then the indices. Terrible. Please do this modification.